airbyte-cdk/python/airbyte_cdk/connector_builder/connector_builder_handler.py

```diff
@@ -14,23 +14,49 @@
 from airbyte_cdk.sources.declarative.declarative_source import DeclarativeSource
 from airbyte_cdk.sources.declarative.declarative_stream import DeclarativeStream
 from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
+from airbyte_cdk.sources.declarative.parsers.model_to_component_factory import ModelToComponentFactory
 from airbyte_cdk.sources.streams.http import HttpStream
 from airbyte_cdk.utils.traced_exception import AirbyteTracedException
 
 DEFAULT_MAXIMUM_NUMBER_OF_PAGES_PER_SLICE = 5
 DEFAULT_MAXIMUM_NUMBER_OF_SLICES = 5
-DEFAULT_MAX_RECORDS = 100
+DEFAULT_MAXIMUM_RECORDS = 100
 
+MAX_PAGES_PER_SLICE_KEY = "max_pages_per_slice"
+MAX_SLICES_KEY = "max_slices"
+MAX_RECORDS_KEY = "max_records"
 
-def read_stream(source: DeclarativeSource, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog) -> AirbyteMessage:
+
+@dataclasses.dataclass
+class TestReadLimits:
+    max_records: int = dataclasses.field(default=DEFAULT_MAXIMUM_RECORDS)
+    max_pages_per_slice: int = dataclasses.field(default=DEFAULT_MAXIMUM_NUMBER_OF_PAGES_PER_SLICE)
+    max_slices: int = dataclasses.field(default=DEFAULT_MAXIMUM_NUMBER_OF_SLICES)
+
+
+def get_limits(config: Mapping[str, Any]) -> TestReadLimits:
+    command_config = config.get("__test_read_config", {})
+    max_pages_per_slice = command_config.get(MAX_PAGES_PER_SLICE_KEY, DEFAULT_MAXIMUM_NUMBER_OF_PAGES_PER_SLICE) or DEFAULT_MAXIMUM_NUMBER_OF_PAGES_PER_SLICE
+    max_slices = command_config.get(MAX_SLICES_KEY, DEFAULT_MAXIMUM_NUMBER_OF_SLICES) or DEFAULT_MAXIMUM_NUMBER_OF_SLICES
+    max_records = command_config.get(MAX_RECORDS_KEY, DEFAULT_MAXIMUM_RECORDS) or DEFAULT_MAXIMUM_RECORDS
+    return TestReadLimits(max_records, max_pages_per_slice, max_slices)
+
+
+def create_source(config: Mapping[str, Any], limits: TestReadLimits) -> ManifestDeclarativeSource:
+    manifest = config.get("__injected_declarative_manifest")
```

Comment:
Nit: I know it predates your code change, but feels like this should be a key lookup instead of `.get`, since the manifest is non-optional as an argument to `ManifestDeclarativeSource`.

Reply:
done